java/arcs/android/storage/database/DatabaseImpl.kt

```diff
@@ -1059,6 +1104,23 @@ class DatabaseImpl(
                     """.trimIndent()
                 clearEntities(nestedEntityQuery, false)
 
+                val nestedEntityCollectionQuery =
```

Comment:
maybe rename the previous to nestedSingletonEntityQuery or add a comment explaining why we need two queries